app/models/school_onboarding.rb

```diff
@@ -71,6 +71,14 @@ def has_event?(event_name)
     events.where(event: event_name).any?
   end
 
+  def last_event(event_name)
+    events.where(event: event_name).last
```

Comment:
probably overkill but does this need some ordering?  e.g. `events.where(event: event_name).order(created_at: :asc).last` (or `events.where(event: event_name).order(updated_at: :asc).last`)

Reply:
Good shout - I did think the same at the same, then I looked at other bits of code in the same class that did the same and saw that it wasn't done there. I think it's a good idea to add it to be sure as you say though!